### PR TITLE
base: remove previous profiles of a member when they're reinvited

### LINKED ISSUE #3277: Invited room members missing display names and avatars
Related to https://github.com/element-hq/element-x-ios/issues/2392

Members fetched through [matrix-sdk.room.members](https://github.com/matrix-org/matrix-rust-sdk/blob/main/crates/matrix-sdk/src/room/mod.rs#L624) (used from [the ffi layer](https://github.com/matrix-org/matrix-rust-sdk/blob/main/bindings/matrix-sdk-ffi/src/room.rs#L184)) and which are in the `Invited` state are missing profiles (display names, avatars).

<del>The problem seems to stem from [this in the sdk-base client](https://github.com/matrix-org/matrix-rust-sdk/blob/main/crates/matrix-sdk-base/src/client.rs#L1174) which intentionally ignores profiles for members which aren't the state event sender.</del>

### PR DESCRIPTION
**Pull request by bnjbvr**
See description commit by commit.

This fixes #3277.

### CODE REVIEW

**Review by stefanceriu — APPROVED** [review]
Can confirm that this does indeed fix the problem Element X side

File: crates/matrix-sdk-base/src/client.rs [hunk 1]
```diff
@@ -338,6 +338,16 @@ impl BaseClient {
                                             .or_default()
                                             .insert(member.sender().to_owned(), member.into());
                                     }
+
+                                    if *member.membership() == MembershipState::Invite {
+                                        // Remove any profile previously stored for the invited
```
Comment (poljar) [inline crates/matrix-sdk-base/src/client.rs:343]: Perhaps it would make sense to explain why we're doing this as well.

**Review by poljar — APPROVED** [review]
Looks good. I left a small nit, thanks a lot for investigating this properly.

File: crates/matrix-sdk-base/src/client.rs (hunk 1 above)
Comment (bnjbvr) [reply crates/matrix-sdk-base/src/client.rs:343]: Good idea, i'll add a small function helper so as to avoid the repetition of code and comments then.

**Pull request merged**